bench-vortex/src/bin/clickbench.rs

```diff
@@ -458,49 +465,49 @@ fn benchmark_duckdb_query(
 }
 
 async fn execute_duckdb_query(
-    query_idx: usize,
     query_string: &str,
-    data_path: &std::path::Path,
+    base_url: &Url,
     file_format: Format,
     single_file: bool,
 ) -> anyhow::Result<Duration> {
-    let query_file = tempfile::tempdir()?
-        .path()
-        .join(format!("query_{query_idx}.sql"));
-
-    fs::write(&query_file, query_string)?;
-
     let extension = match file_format {
         Format::Parquet => "parquet",
         Format::OnDiskVortex => "vortex",
         other => vortex_panic!("Format {other} isn't supported on ClickBench"),
     };
 
+    // Base path contains trailing /.
     let file_glob = if single_file {
-        format!(
-            "{}/{extension}/hits.{extension}",
-            data_path.to_string_lossy()
-        )
+        format!("{base_url}{extension}/hits.{extension}")
     } else {
-        format!("{}/{extension}/*.{extension}", data_path.to_string_lossy())
+        format!("{base_url}{extension}/*.{extension}")
     };
 
+    // TODO: double check
+    let file_glob = file_glob.strip_prefix("file://").unwrap();
     let time_instant = Instant::now();
+    let register_tables =
+        format!("CREATE VIEW hits AS SELECT * FROM read_{extension}('{file_glob}')",);
+
+    let repo_root = std::process::Command::new("git")
+        .args(["rev-parse", "--show-toplevel"])
+        .output()
+        .map(|output| String::from_utf8_lossy(&output.stdout).trim().to_string())
+        .unwrap_or_else(|_| ".".to_string());
+
+    let duckdb_path = format!("{repo_root}/duckdb-vortex/build/release/duckdb");
 
-    // TODO: complete duckdb setup
-    let output = tokio::process::Command::new("duckdb")
-        // Create a temporary database in RAM.
-        .arg(":memory:")
+    let output = tokio::process::Command::new(&duckdb_path)
```

Comment:
Overall, I'm leaning towards using [duckdb-rs](https://github.com/duckdb/duckdb-rs) to run multiple subsequent commands. Importantly, I want to register the tables first before starting the benchmark timer.

Reply:
There's 2 things I don't like about the current draft:
- how we find the duckdb binary
- table registration is included in the benchmark, unlike us handling that separately upfront in datafusion